independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -121,6 +121,24 @@ public void destroy(Contextual<?> contextual) {
         }
     }
 
+    @Override
+    public ContextState activateInitial() {
+        if (LOG.isTraceEnabled()) {
+            String stack = Arrays.stream(Thread.currentThread().getStackTrace())
+                    .skip(2)
+                    .limit(7)
+                    .map(se -> "\n\t" + se.toString())
+                    .collect(Collectors.joining());
+            LOG.tracef("Activate %s %s\n\t...", "new", stack);
+        }
+        var state = new RequestContextState(new ConcurrentHashMap<>());
+        assert currentContext.get() == null;
```

Comment:
@Ladicek @geoand hope this assertion won't mess up on Tests, but probably worth make tests to run and if it breaks something. In short: can be used to detect racy writes and/or earlier initial activations

Reply:
This assert actually breaks one of the TCK tests but I think that our `io.quarkus.arc.tck.porting.ContextsImpl` needs to be fixed - see my second commit.